src/components/Forum/ForumApprovedPostDetails/ForumApprovedPostDetails.js

```diff
@@ -0,0 +1,103 @@
+import { LinearProgress, makeStyles, Typography } from '@material-ui/core';
+import firebase from 'firebase/app';
+import 'firebase/firestore';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router';
+import ForumBreadcrumbs from './ForumApprovedPostBreadcrumbs';
+import ForumPostDetailsList from './ForumPostDetailsList';
+
+const useStyles = makeStyles((theme) => ({
+  navigationContainer: {
+    marginBottom: theme.spacing(4),
+  },
+}));
+
+export default function ForumApprovedPostDetails() {
+  const classes = useStyles();
+  const { postID } = useParams();
+
+  const [user, setUser] = useState({});
+  const [userLoading, setUserLoading] = useState(false);
+  const [userError, setUserError] = useState(null);
+
+  const [category, setCategory] = useState({});
+  const [categoryLoading, setCategoryLoading] = useState(false);
+  const [categoryError, setCategoryError] = useState(null);
+
+  const [post, setPost] = useState({});
+  const [postLoading, setPostLoading] = useState(false);
+  const [postError, setPostError] = useState(null);
+
+  useEffect(() => {
+    setPostLoading(true);
+    firebase.firestore().collection('forum_posts').doc(postID).get()
+      .then(async (snapshot) => {
```

Comment:
Is making this function a promise with async necessary here? Do we need to await anything before we set the post hook?

Reply:
My bad, it's not needed.